src/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -39,6 +37,12 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
             }
         }
 
+        protected internal override void BeforeRequest<TRequest>(TRequest request)
+        {
+            // Update the locale for this request to the desired LSP locale.
+            CultureInfo.CurrentUICulture = GetCultureForRequest();
```

Comment:
Fixing a bug that showed up after the queue changes - previously we were running this in WrapStartRequestTaskAsync, but the task could potentially be completed by the time `WrapStartRequestTaskAsync` was called.  Instead we need to explicitly set this before the request task runs.